src/util/tubemap.test.js

```diff
@@ -211,4 +214,394 @@ describe('cigar_string', () => {
+                "nodeName": "1",
+                "mismatches": []
+            },
+            {
+                "nodeName": "12",
+                "mismatches": []
+            },
+            {
+                "nodeName": "13",
+                "mismatches": []
+            }
+        ],
+        "firstNodeOffset": 0,
+        "finalNodeCoverLength": 1,
+      }
+    ];
+    expect(checkNodeExample(node, reads)).toBe(undefined);
+    expect(coverage(node, reads)).toBe(1.00);
+  })
+  // TEST #3
+  it('can handle node of length of 6 with 2 outgoing reads and 2 internal reads', async () => {
+    const node = {        
+      nodename: '3',    
+      sequenceLength: 6,
+      incomingReads: [],
+      internalReads: [0, 1],
+      outgoingReads: [[2, 0], [3, 0]]
+    }
+    const reads = [
+      {
+        "id": 1,
+        "sequenceNew": [
+          {
+            "nodeName": "1",
+            "mismatches": [
+              {
+                "type": "insertion",
+                "pos": 0,
+                "seq": "CACAG"
+              }
+            ]
+          }
+        ],
+        "firstNodeOffset": 0,
+        "finalNodeCoverLength": 3,
+      },
+      {
+        "id": 2,
+        "sequenceNew": [
+          {
+            "nodeName": "1",
+            "mismatches": [
+              {
+                "type": "insertion",
+                "pos": 0,
+                "seq": "TCAAGAACAGTCATTCATG"
+              }
+            ]
+          }
+        ],
+        "firstNodeOffset": 1,
+        "finalNodeCoverLength": 5,
+      }, 
+      {
+        "id": 3,
+        "sequenceNew": [
+            {
+              "nodeName": "1",
+              "mismatches": []
+            },
+            {
+              "nodeName": "11",
+              "mismatches": []
+            },
+            {
+              "nodeName": "13",
+              "mismatches": []
+            }
+        ],
+        "firstNodeOffset": 2,
+        "finalNodeCoverLength": 6,
+      },
+      {
+        "id": 4,
+        "sourceTrackID": "1",
+        "sequence": [
+          "1",
+          "12",
+          "13"
+        ],
+        "sequenceNew": [
+            {
+              "nodeName": "1",
+              "mismatches": []
+            },
+            {
+              "nodeName": "12",
+              "mismatches": []
+            },
+            {
+              "nodeName": "13",
+              "mismatches": []
+            }
+        ],
+        "firstNodeOffset": 4,
+        "finalNodeCoverLength": 6,
+      }
+    ];
+    expect(checkNodeExample(node, reads)).toBe(undefined);
+    expect(coverage(node, reads)).toBe(2.17);
+  })
+  // TEST #4
+  it('can handle node of length of 30 with 2 incoming reads, 4 internal reads, and 3 outgoing reads', async () => {
+    const node = {        
+      nodename: '4',    
+      sequenceLength: 30,
+      incomingReads: [[0, 1], [1, 1]],
+      internalReads: [2, 3, 4, 5],
+      outgoingReads: [[6, 2], [7, 2], [8, 2]],
```

Comment:
If outgoing reads are all meant to start in the nodes they are outgoing from, why are these reads listed as outgoing for this node but at visit index 2?